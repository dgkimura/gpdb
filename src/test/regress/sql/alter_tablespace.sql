-- Given we create a database
DROP DATABASE IF EXISTS mydatabase;
CREATE DATABASE mydatabase;
\c mydatabase

-- And create a table in that database with data
DROP TABLE IF EXISTS mytable;
CREATE TABLE mytable(id int, name text);
INSERT INTO mytable VALUES(1, 'a');

-- When we create a tablespace
DROP TABLESPACE IF EXISTS mytablespace;
\! rm -rf '/tmp/mytablespace';
\! mkdir '/tmp/mytablespace';
CREATE TABLESPACE mytablespace LOCATION '/tmp/mytablespace';

-- And alter the database to use the new tablespace
\c postgres
ALTER DATABASE mydatabase SET TABLESPACE mytablespace;

-- Then all the database files from QD and all QEs should be moved into the new tablespace. We use the fact that default tablespace names look like base/<OID>/<TABLE OID> wherease user created tablespaces have format pg_tblspc/<other stuff>
\c mydatabase
SELECT COUNT(*) FROM (SELECT pg_relation_filepath('mytable')) a where a.pg_relation_filepath LIKE '%pg_tblspc%';
SELECT COUNT(*) FROM (SELECT gp_segment_id, pg_relation_filepath('mytable') FROM gp_dist_random('gp_id')) a where a.pg_relation_filepath LIKE '%pg_tblspc%';

