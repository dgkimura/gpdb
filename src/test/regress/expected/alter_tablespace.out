-- Given we create a database
DROP DATABASE IF EXISTS mydatabase;
CREATE DATABASE mydatabase;
\c mydatabase
-- And create a table in that database with data
DROP TABLE IF EXISTS mytable;
NOTICE:  table "mytable" does not exist, skipping
CREATE TABLE mytable(id int, name text);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'id' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO mytable VALUES(1, 'a');
-- When we create a tablespace
DROP TABLESPACE IF EXISTS mytablespace;
\! rm -rf '/tmp/mytablespace';
\! mkdir '/tmp/mytablespace';
CREATE TABLESPACE mytablespace LOCATION '/tmp/mytablespace';
-- And alter the database to use the new tablespace
\c postgres
ALTER DATABASE mydatabase SET TABLESPACE mytablespace;
-- Then all the database files from QD and all QEs should be moved into the new tablespace. We use the fact that default tablespace names look like base/<OID>/<TABLE OID> wherease user created tablespaces have format pg_tblspc/<other stuff>
\c mydatabase
SELECT COUNT(*) FROM (SELECT pg_relation_filepath('mytable')) a where a.pg_relation_filepath LIKE '%pg_tblspc%';
 count 
-------
     1
(1 row)

SELECT COUNT(*) FROM (SELECT gp_segment_id, pg_relation_filepath('mytable') FROM gp_dist_random('gp_id')) a where a.pg_relation_filepath LIKE '%pg_tblspc%';
 count 
-------
     3
(1 row)

